armory/metrics/task.py

```diff
@@ -548,25 +566,236 @@ def _intersection_over_union(box_1, box_2):
     ):
         log.warning("One set of boxes appears to be normalized while the other is not")
 
-    # Determine coordinates of intersection box
-    x_left = max(box_1[1], box_2[1])
-    x_right = min(box_1[3], box_2[3])
-    y_top = max(box_1[0], box_2[0])
-    y_bottom = min(box_1[2], box_2[2])
-
-    intersect_area = max(0, x_right - x_left) * max(0, y_bottom - y_top)
+    intersect_area = _intersection(box_1, box_2)
     if intersect_area == 0:
         return 0
 
-    box_1_area = (box_1[3] - box_1[1]) * (box_1[2] - box_1[0])
-    box_2_area = (box_2[3] - box_2[1]) * (box_2[2] - box_2[0])
+    union_area = _union(box_1, box_2)
 
-    iou = intersect_area / (box_1_area + box_2_area - intersect_area)
+    iou = intersect_area / union_area
     assert iou >= 0
     assert iou <= 1
     return iou
 
 
+def _area_of_polygon(points):
+    """
+    Shoelace formula for area of a polygon.
+    https://en.wikipedia.org/wiki/Shoelace_formula
+
+    points is an array or list of shape (N, 2) and points may be in clockwise or counterclockwise order
+    """
+
+    if type(points) == list:
```

Comment:
Addressing https://github.com/twosixlabs/armory/actions/runs/5788881976/job/15688781545?pr=1946:
```suggestion
    if isinstance(points, list):
```